AhmedFatthy1040/hias
# Setup Husky and Commitlint for Commit Message Validation

### **Title**  
Setup Husky and Commitlint for Commit Message Validation  

---

### **Description**  
This PR introduces Husky and Commitlint to the project, ensuring all commit messages follow conventional commit standards for better readability and consistency.  

---

### **Changes Made**  
- Installed **Husky** for managing Git hooks.  
- Added a `pre-commit` hook with a placeholder script for future customization.  
- Configured a `commit-msg` hook to validate commit messages using **Commitlint**.  
- Created a basic Commitlint configuration using `@commitlint/config-conventional`.  

---

### **How to Test**  
1. Pull this branch locally.  
2. Make a new commit:  
   - Valid commit message: `git commit -m "feat: add feature"`.  
   - Invalid commit message: `git commit -m "invalid message"`.  
3. Ensure invalid messages are rejected and valid ones proceed as expected.  

---

### **Why This Change?**  
Adding Husky and Commitlint helps enforce a consistent commit message structure, which improves collaboration, release notes generation, and overall codebase maintainability.  

---

### **Checklist**  
- [x] Husky installed and initialized.  
- [x] Commitlint installed and configured.  
- [x] Pre-commit hook set up.  
- [x] Commit-msg hook for validation added.  

---

Feel free to suggest additional improvements or ask questions! 😊
